yt_dlp/extractor/bilibili.py

```diff
@@ -139,283 +128,268 @@ class BiliBiliIE(InfoExtractor):
         },
     }]
 
-    _APP_KEY = 'iVGUTjsxvpLeuDCf'
-    _BILIBILI_KEY = 'aHRmhWMLkdeMuILqORnYZocwMBpMEOdt'
-
-    def _report_error(self, result):
-        if 'message' in result:
-            raise ExtractorError('%s said: %s' % (self.IE_NAME, result['message']), expected=True)
-        elif 'code' in result:
-            raise ExtractorError('%s returns error %d' % (self.IE_NAME, result['code']), expected=True)
-        else:
-            raise ExtractorError('Can\'t extract Bangumi episode ID')
+    def json2srt(self, json_data):
+        srt_data = ''
+        for idx, line in enumerate(json_data.get('body', [])):
+            srt_data += f'{idx + 1}\n'
+            srt_data += f'{srt_subtitles_timecode(line["from"])} --> {srt_subtitles_timecode(line["to"])}\n'
+            srt_data += f'{line["content"]}\n\n'
+        return srt_data
 
     def _real_extract(self, url):
-        url, smuggled_data = unsmuggle_url(url, {})
-
         mobj = self._match_valid_url(url)
         video_id = mobj.group('id_bv') or mobj.group('id')
 
-        av_id, bv_id = self._get_video_id_set(video_id, mobj.group('id_bv') is not None)
-        video_id = av_id
-
-        info = {}
-        anime_id = mobj.group('anime_id')
-        page_id = mobj.group('page')
         webpage = self._download_webpage(url, video_id)
 
-        # Bilibili anthologies are similar to playlists but all videos share the same video ID as the anthology itself.
-        # If the video has no page argument, check to see if it's an anthology
-        if page_id is None:
+        if '开通大会员观看' in webpage and '__playinfo__' not in webpage:
+            raise ExtractorError(f'VIP is required for {url}', expected=True)
+
+        initial_state = self._search_json(r'window.__INITIAL_STATE__\s*=\s*', webpage, 'initial state', video_id)
+
+        is_bangumi = mobj.group('bangumi') is not None
+        video_data = initial_state.get('epInfo') or initial_state.get('videoData') or {}
+        if 'bvid' not in video_data:
+            raise ExtractorError(f'Unknown webpage json schema{bug_reports_message()}')
```

Comment:
```suggestion
        video_data = traverse_obj(initial_state, 'epInfo', 'videoData') or {}
```
The existing error message is sufficient